src/services/juno/custom/helpers/admin-resource.ts

```diff
@@ -0,0 +1,81 @@
+import { AdminResources } from "services/types";
+import { CW3Config } from "types/contracts";
+import { queryContract } from "services/juno/queryContract";
+import { isJunoAddress } from "schemas/tests";
+import { contracts } from "constants/contracts";
+import { adminRoutes, appRoutes } from "constants/routes";
+import { defaultProposalTags } from "../../tags";
+import { customApi } from "../custom";
+
+export const AP_ID = 0;
+export const REVIEWER_ID = 0.5;
+
+export function getCWs(id: number) {
+  //TODO: atm, only two admin types, refactor this once > 2
+  //charities doesn't have hardcoded cws, so only test for AP_ID && REVIEWER_ID
+  const cw3 = id === AP_ID ? contracts.cw3ApTeam : contracts.cw3ReviewTeam;
+  const cw4 =
+    id === AP_ID ? contracts.cw4GrpApTeam : contracts.cw4GrpReviewTeam;
+  const type: AdminResources["type"] = id === AP_ID ? "ap" : "review";
+  return { cw3, cw4, type };
+}
+
+export function isAp(id: number) {
+  return id === AP_ID || id === REVIEWER_ID;
+}
+
+export async function getMeta(
+  endowId: number,
+  cw3: string,
+  user?: string
+): Promise<[AdminResources["propMeta"], CW3Config]> {
+  const [votersRes, config, voter] = await Promise.all([
+    queryContract("cw3ListVoters", cw3, null),
+    queryContract("cw3Config", cw3, null),
+    /** just set credential to none, if disconnected or non-juno wallet */
+    user && isJunoAddress(user)
+      ? queryContract("cw3Voter", cw3, {
+          addr: user,
+        })
+      : Promise.resolve({ weight: 0 }),
+  ]);
+
+  const numVoters = votersRes.voters.length;
+
+  const willExecute =
+    /** single member */
+    numVoters === 1 ||
+    /** multiple members but threshold is lte 1/members given that execution is not required */
+    (!config.require_execution &&
+      Number(config.threshold.absolute_percentage.percentage) <= 1 / numVoters);
+
+  const tagPayloads = [customApi.util.invalidateTags(defaultProposalTags)];
+
+  const meta = willExecute
```

Comment:
Suggestion: always better to split huge ternary operator segments into functions or chunks
eg:
const url = willExecute ? `${appRoutes.admin}/${endowId}` : `${appRoutes.admin}/${endowId}/${proposals}`;
const description = willExecute ? "Go to admin home" : "Go to proposals";
const message = willExecute ? "Successful transaction" : "Proposal successfully created";
const meta = {
  willExecute,
  successMeta: { message, link: { url, description } },
  tagPayloads,
  isAuthorized: Boolean(weight),
};

Reply:
@abbasali1994, 
these are nice suggestions - though not the essence of this PR (forgot to mention in PR description that I just moved these helpers since `custom.ts` is getting long )

on the otherhand I created `backlog` for this refactors
https://github.com/orgs/AngelProtocolFinance/projects/2/views/2?pane=issue&itemId=20653145